src/main/java/org/mockito/verification/JUnitVerificationCollector.java

```diff
@@ -0,0 +1,108 @@
+package org.mockito.verification;
+
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+import org.mockito.exceptions.base.MockitoAssertionError;
+import org.mockito.internal.MockitoCore;
+import org.mockito.internal.verification.api.VerificationData;
+
+import static org.mockito.Mockito.times;
+
+/**
+ * Use this rule in order to collect multiple verification failures and report at once.
+ *
+ * In the example below, the verification failure thrown by {@code byteReturningMethod()} does not block
+ * verifying against the {@code simpleMethod()}. After the test is run, a report is generated stating all
+ * collect verification failures.
+ *
+ * <pre class="code"><code class="java">
+ *   &#064;Rule
+ *   public JUnitVerificationCollector collector = new JUnitVerificationCollector();
```

Comment:
I think this rule should be instantiated this way (Also the method should only expose an interface for example `VerificationCollector`) :

```java
@Rule VerificationCollector collector = MockitoJUnit.collector();
```